pkg/crc/preset/preset.go

```diff
@@ -15,16 +15,20 @@ const (
 	Microshift Preset = "microshift"
 )
 
+func presetMap() map[Preset]string {
+	return map[Preset]string{
+		Podman:     string(Podman),
+		OpenShift:  string(OpenShift),
+		OKD:        string(OKD),
+		Microshift: string(Microshift),
+	}
+}
+
 func (preset Preset) String() string {
-	switch preset {
-	case Podman:
-		return string(Podman)
-	case OpenShift:
-		return string(OpenShift)
-	case OKD:
-		return string(OKD)
-	case Microshift:
-		return string(Microshift)
+	for pSet, pString := range presetMap() {
+		if pSet == preset {
+			return pString
+		}
 	}
```

Comment:
you have a map, you don't need to iterate over it
```
presetStr, presetExists := presetMap()[preset]
if !presetExists {
    return "invalid"
}
return presetStr
```

Reply:
updated